ttnn/cpp/ttnn/operations/moreh/moreh_dot_op_backward/device/moreh_dot_backward_device_operation.hpp

```diff
@@ -21,12 +21,13 @@ struct MorehDotBackwardOperation {
         const Tensor &output_grad;
         const Tensor &input;
         const Tensor &other;
-        const std::optional<const Tensor> &input_grad;
-        const std::optional<const Tensor> &other_grad;
+
+        // (o2buzzle): May I present: thanhnguyen's mistake that costed me 3 hours.
+        const std::vector<std::optional<Tensor>> output_tensors;
```

Comment:
Why is it optional if this operation only works inplace?
Also, why does this operation only works inplace?

Reply:
That is what was in the code when I was working on porting the old tt-eager code to the new one. It was not supposed to create a new tensor: 
![image](https://github.com/user-attachments/assets/ad220608-1f11-4595-9c40-91d45aded5ab)

So the ported code pretty much replicated this behavior

Reply:
Also please keep in mind that the current version in the tree was not ever correct (but no test failed because when it was merged, it ran the old `tt_eager` version for the tests instead of the newly changed code)

Reply:
Thank you for the context.